pjdfstest.c

```diff
@@ -96,8 +96,10 @@
 #define	HAS_FSTATAT
 #define	HAS_LINKAT
 #define	HAS_MKDIRAT
+#ifndef	__APPLE__
```

Comment:
This nests the OSX defines within the "other" list.  I think it would be more clear if the defines went one OS at a time, eg
```
#ifdef __FreeBSD_version
...
#else ifdef __APPLE__
...
# else //all others
...
#endif
```
Though really, I think autoconf would be preferable.